postgresql11/plan.sh

```diff
@@ -0,0 +1,140 @@
+# shellcheck disable=SC2164
+pkg_name=postgresql11
+pkg_version=11.1
+pkg_origin=core
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_description="PostgreSQL is a powerful, open source object-relational database system."
+pkg_upstream_url="https://www.postgresql.org/"
+pkg_license=('PostgreSQL')
+pkg_source="https://ftp.postgresql.org/pub/source/v${pkg_version}/postgresql-${pkg_version}.tar.bz2"
+pkg_shasum="90815e812874831e9a4bf6e1136bf73bc2c5a0464ef142e2dfea40cda206db08"
+pkg_dirname="postgresql-${pkg_version}"
+
+pkg_deps=(
+  core/bash
+  core/gawk
+  core/glibc
+  core/grep
+  core/libossp-uuid
+  core/openssl
+  core/perl
+  core/readline
+  core/zlib
+
+  # for postgis
+  core/libxml2
+  core/geos
+  core/proj
+  core/gdal
+)
+
+pkg_build_deps=(
+  core/coreutils
+  core/gcc
+  core/make
+
+  # for postgis
+  core/diffutils
+)
+
+pkg_bin_dirs=(bin)
+pkg_include_dirs=(include)
+pkg_lib_dirs=(lib)
+pkg_exports=(
+  [port]=port
+  [superuser_name]=superuser.name
+  [superuser_password]=superuser.password
+)
+pkg_exposes=(port)
+
+ext_postgis_version=2.5.1
```

Comment:
I wonder if we want to consider removing postgis from the pg11 package.   🤔  

It certainly makes things easier for folks who need spatial support in Postgresql and we have precedent from the pg9x packages, but there are a drawbacks as well.  It roughly doubles the install footprint, which is not delightful if you don't need spatial support. It adds to the complexity of maintenance of this piece of software, and doesn't provide a pattern for other extensions a consumer might need. We also recently had a discussion about bundling libraries in packages in a BATS PR #2287 that it was decided to not follow this pattern. 

@irvingpop @stevendanna Thoughts?

Reply:
Yeah I think you're right @smacfarlane -   I like the pattern we established in the `sqitch_pg` plan and that's a lot cleaner.   I think this plan is the right time and place to make the split.  

When somebody wants postgis in the future, it'll make sense to make a `postgresql11-postgis` plan that adds these bits back in.